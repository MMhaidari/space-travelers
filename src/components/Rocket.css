.rocket {
  display: flex;
  gap: 1.4rem;
  margin: 2rem 5rem 4rem 5rem;
}

.rocket div:nth-child(2) {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  gap: 2rem;
}

.rocket-head {
  margin-top: 2rem;
}

.rocket-text {
  font-size: 1.4rem;
}

img {
  width: 30rem;
}

.reserve-btn {
  background-color: #007bff;
  color: #fff;
  border: none;
  padding: 1rem 2rem;
  border-radius: 0.4rem;
  font-size: 1.6rem;
  cursor: pointer;
}

.reserve-Btn:hover {
  background-color: #0069d9;
}

.cancel-btn {
  background-color: #ecf2f8;
  color: #5c5c5c;
  border: none;
  padding: 1rem 2rem;
  border-radius: 0.4rem;
  font-size: 1.6rem;
  cursor: pointer;
}

.cancel-btn:hover {
  background-color: #0069d9;
}

.hidden {
  display: none;
}

.show {
  background-color: #519be9;
  border: none;
  color: #fff;
  border-radius: 0.4rem;
  margin-right: 0.6rem;
  padding: 0.5rem;
}