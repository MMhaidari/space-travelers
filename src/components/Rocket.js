import React from 'react';
import PropTypes from 'prop-types';
import './Rocket.css';

const Rocket = ({ rocket }) => (
  <section>
    <div className="rocket">
      <div>
        {rocket.flickr_images && rocket.flickr_images.length > 0 && (
        <img src={rocket.flickr_images[0]} alt={`Rocket ${rocket.name}`} />
        )}
      </div>
      <div>
        <h2 className="rocket-head">{rocket.name}</h2>
        <p className="rocket-text">
          <span className={rocket.reserved === true ? 'show' : 'hidden'}>Reserved</span>
          {rocket.description}
        </p>
        {!rocket.reserved && (
        <button
          type="button"
          className="reserve-btn"
        >
          Reserve Rocket
        </button>
        )}
        {rocket.reserved && (
        <button
          type="button"
          className="cancel-btn"
        >
          Cancel Reservation
        </button>
        )}
      </div>
    </div>
  </section>
);

Rocket.propTypes = {
  rocket: PropTypes.oneOfType([PropTypes.object]).isRequired,
};

export default Rocket;
