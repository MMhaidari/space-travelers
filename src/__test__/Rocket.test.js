import React from 'react';
import renderer from 'react-test-renderer';
import Rocket from '../components/Rocket';

describe('Rocket component', () => {
  const rocket = {
    id: 'falcon1',
    name: 'Falcon 1',
    description: 'Small launch vehicle',
    flickr_images: ['https://www.example.com/falcon1.jpg'],
    reserved: true,
  };

  test('renders when rocket is reserved', () => {
    const tree = renderer
      .create(<Rocket rocket={rocket} handleReserveClick={() => {}} onCancelClick={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders correctly when rocket is not reserved', () => {
    const newRocket = { ...rocket, reserved: false };
    const tree = renderer
      .create(<Rocket rocket={newRocket} handleReserveClick={() => {}} onCancelClick={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
