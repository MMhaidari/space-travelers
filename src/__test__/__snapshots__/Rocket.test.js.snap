// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket component renders correctly when rocket is not reserved 1`] = `
<section>
  <div
    className="rocket"
  >
    <div>
      <img
        alt="Rocket Falcon 1"
        src="https://www.example.com/falcon1.jpg"
      />
    </div>
    <div>
      <h2
        className="rocket-head"
      >
        Falcon 1
      </h2>
      <p
        className="rocket-text"
      >
        <span
          className="hidden"
        >
          Reserved
        </span>
        Small launch vehicle
      </p>
      <button
        className="reserve-btn"
        onClick={[Function]}
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
</section>
`;

exports[`Rocket component renders correctly when rocket is reserved 1`] = `
<section>
  <div
    className="rocket"
  >
    <div>
      <img
        alt="Rocket Falcon 1"
        src="https://www.example.com/falcon1.jpg"
      />
    </div>
    <div>
      <h2
        className="rocket-head"
      >
        Falcon 1
      </h2>
      <p
        className="rocket-text"
      >
        <span
          className="show"
        >
          Reserved
        </span>
        Small launch vehicle
      </p>
      <button
        className="cancel-btn"
        onClick={[Function]}
        type="button"
      >
        Cancel Reservation
      </button>
    </div>
  </div>
</section>
`;

exports[`Rocket component renders when rocket is reserved 1`] = `
<section>
  <div
    className="rocket"
  >
    <div>
      <img
        alt="Rocket Falcon 1"
        src="https://www.example.com/falcon1.jpg"
      />
    </div>
    <div>
      <h2
        className="rocket-head"
      >
        Falcon 1
      </h2>
      <p
        className="rocket-text"
      >
        <span
          className="show"
        >
          Reserved
        </span>
        Small launch vehicle
      </p>
      <button
        className="cancel-btn"
        onClick={[Function]}
        type="button"
      >
        Cancel Reservation
      </button>
    </div>
  </div>
</section>
`;
